backend/util/postUtil.js

```diff
@@ -64,11 +64,43 @@ var tagExists = async function(tag,projectId){
 }
 
 
+var homepagePosts = async function(userId){
+    user_interests = await UserInterest.findAll({
+	where: {
+	    user_id : userId
+	},
+	attributes : ['interest']
+    });
+    interest_array = []
+    for(var i in user_interests)
```

Comment:
You can use .map() for that, for this instance:

interest_array = user_interests.map( e => e.interest)

That would simplify it by a lot, and is a much more modern approach

Reply:
thanks for recommendation